src/Products/Comparison/Service/ComparisonServiceImpl.cs

```diff
@@ -92,178 +95,206 @@ public bool CheckFiles(CompareRequest files)
                 {
-                    resultImages[pageNumber - 1].PageStream.CopyTo(ms);
-                    bytes = ms.ToArray();
-                }
-                loadedPage.SetData(Convert.ToBase64String(bytes));
-                loadedPage.number = pageNumber;
-                loadedPage.height = resultImages[pageNumber - 1].Height;
-                loadedPage.width = resultImages[pageNumber - 1].Width;
+                    using (Comparer comparer = new Comparer(fileStream, GetLoadOptions(password)))
+                    {
+
+                        byte[] bytes = RenderPageToMemoryStream(comparer, pageNumber - 1).ToArray();
+                        string encodedImage = Convert.ToBase64String(bytes);
+                        loadedPage.SetData(encodedImage);
 
+                        loadedPage.number = pageNumber;
+                        loadedPage.height = 842;
+                        loadedPage.width = 595;
+                    }
+                }
             }
             catch (Exception ex)
             {
                 throw new FileLoadException("Exception occurred while loading result page", ex);
             }
+
             return loadedPage;
         }
 
-        public LoadDocumentEntity LoadDocumentInfo(PostedDataEntity postedData)
+        static MemoryStream RenderPageToMemoryStream(Comparer comparer, int pageNumberToRender)
         {
-            LoadDocumentEntity loadDocumentEntity = new LoadDocumentEntity();
+            MemoryStream result = new MemoryStream();
 
-            string password = "";
-            try
+            PreviewOptions previewOptions = new PreviewOptions(pageNumber => result)
             {
-                // get/set parameters
-                string documentGuid = postedData.guid;
-                password = (String.IsNullOrEmpty(postedData.password)) ? null : postedData.password;
-                Comparer comparer = new Comparer();
-                List<PageImage> resultImages = comparer.ConvertToImages(documentGuid, password);
+                PreviewFormat = PreviewFormats.PNG,
+                PageNumbers = new[] { pageNumberToRender },
+                ReleasePageStream = UserReleaseStreamMethod
+            };
 
-                foreach (PageImage page in resultImages)
-                {
-                    PageDescriptionEntity loadedPage = new PageDescriptionEntity();
-                    loadDocumentEntity.SetPages(loadedPage);
-                }
-                return loadDocumentEntity;
-            }
-            catch (Exception ex)
-            {
-                // set exception message
-                throw new FileLoadException("Exception occurred while loading document info", ex);
-            }
+            comparer.Source.GeneratePreview(previewOptions);
+
+            return result;
         }
 
+        private static void UserReleaseStreamMethod(int pageNumber, Stream stream)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'pageNumber'.](https://app.codacy.com/gh/groupdocs-comparison/GroupDocs.Comparison-for-.NET-MVC/pullRequest?prid=5354506)